Improvements for database handle and connection attributes

pali opened the pull request:
* Allow to modify `$dbh->{mariadb_multi_statements}` also after connection to server was established
* Allow to specify `mariadb_auth_plugin` connection attribute

choroba commented on lib/DBD/MariaDB.pod:
```diff
@@ -133,6 +133,10 @@ This is similar to the behavior of the C<mariadb> or C<mysql> command line
 client. Also, C<SELECT DATABASE()> will return the current database active for
 the handle.
 
+The DBD::MariaDB driver understand any connection attribute specified in
```
understand<ins>s</ins>

pali replied:
Fixed.

choroba commented on dbdimp.h:
```diff
@@ -537,6 +537,7 @@ struct imp_dbh_st {
                                */
     bool use_server_side_prepare;
     bool disable_fallback_for_server_prepare;
+    bool use_multi_statements;
```
Any plans to add a test for the new behaviour?

pali replied:
I added a new test which changes `$dbh->{mariadb_multi_statements}` at runtime.

choroba commented on dbdimp.c:
```diff
@@ -1855,6 +1855,29 @@ static bool mariadb_dr_connect(
+            return FALSE;
+          }
+        }
+
```
Again, what about testing?

pali replied:
Fully automated testing of `mariadb_auth_plugin` is not possible. It requires configuration of _different_ authentication method which would be understood by both client and server. Maybe it could be possible to write github CI testing for it, but that would require to change / hack how MySQL::Sandbox setup mysql server in its virtual env.

choroba replied:
Not even the `mysql_native_password` with a new client?

pali replied:
We cannot expect that `mysql_native_password` is always present in new mysql versions (e.g. system admin can decide to not install support for it) and also we cannot expect that oracle does not decide to drop support for this specific plugin in some new mysql version. So automated test which is always run during install/test phase of perl DBD-MariaDB and uses `mysql_native_password` is fragile and can cause breaking of installation in future.

We can run such test only in controlled env, e.g. on github actions under configuration where we are sure that this one specific method `mysql_native_password` is enabled and is working.

pali commented:
Unfortunately new tests for `mariadb_auth_plugin` are failing (bug in MySQL 8.x client), details in: https://github.com/perl5-dbi/DBD-MariaDB/issues/156#issuecomment-1646844330

So I would have to prepare different test for `mariadb_auth_plugin`.

pali commented:
Unfortunately I was not able to write any other reliable automated test for `mariadb_auth_plugin` due to those mentioned issues.

I squashed commits and dropped changes for non-working tests.
